### ISSUE TITLE
GCC internal compiler error

### ISSUE THREAD

**Issue by rsp** [rsp/c-experiments#1]
I got an internal GCC error in commit 89f3d27cfc84e7e24c8a8ae05f1e23d205888f4c using flags from https://stackoverflow.com/questions/10551665/how-to-create-4kb-linux-binaries#10552160 by @phresnel

```
gcc empty.c -o empty-phresnel -Os -mfpmath=387 \
    -mfancy-math-387 -fmerge-all-constants -fsingle-precision-constant \
    -fno-math-errno -Wall -ldl -ffast-math -nostartfiles -nostdlib  \
    -fno-unroll-loops -fshort-double
<built-in>:0:0: internal compiler error: in layout_type, at stor-layout.c:2141
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
Preprocessed source stored into /tmp/ccueDNUj.out file, please attach this to your bugreport.
make: *** [empty-phresnel] Error 1
```

**Comment by rsp** [comment]
I have to see what is causing it, which flag and if it can be reproduced on other gcc versions.

**Comment by rsp** [comment]
Output of test script from commit 8f34b05decbd4e297a6db156750c939ec562ad0f:

```
$ ./test-flags 
TESTING FLAG: -Os
OK
TESTING FLAG: -mfpmath=387
OK
TESTING FLAG: -mfancy-math-387
OK
TESTING FLAG: -fmerge-all-constants
OK
TESTING FLAG: -fsingle-precision-constant
OK
TESTING FLAG: -fno-math-errno
OK
TESTING FLAG: -Wall
OK
TESTING FLAG: -ldl
OK
TESTING FLAG: -ffast-math
OK
TESTING FLAG: -nostartfiles
/usr/bin/ld: warning: cannot find entry symbol _start; defaulting to 00000000004002a3
OK
TESTING FLAG: -nostdlib
/usr/bin/ld: warning: cannot find entry symbol _start; defaulting to 0000000000400144
OK
TESTING FLAG: -fno-unroll-loops
OK
TESTING FLAG: -fshort-double
<built-in>:0:0: internal compiler error: in layout_type, at stor-layout.c:2141
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
Preprocessed source stored into /tmp/ccJ3n69U.out file, please attach this to your bugreport.
ERROR
```

**Comment by rsp** [comment]
Ok, it's the -fshort-double flag that's crashing gcc.
-nostartfiles and -nostdlib are causing warnings.

Found more info - by Andi Kleen on https://gcc.gnu.org/ml/gcc-patches/2014-09/msg02464.html

> -fshort-double has crashes the compiler since 4.6 (see PR60410)
> Since it's an obscure option that apparently nobody uses it the
> best way to fix it seems to just remove it.

**Comment by rsp** [comment]
Closed by commit 8f22b3176eea89726569f6a4b1798889c4b3270b
